### ISSUE TITLE
In chrome os the referrer is empty

### ISSUE THREAD

**Issue by alex2844** [GoogleChrome/android-browser-helper#64]
chrome os: 80.0.3987.76 beta
androidbrowserhelper: 1.1.0

**Comment by alex2844** [comment]
https://github.com/GoogleChromeLabs/svgomg-twa/issues/72

**Comment by andreban** [comment]
@dominickng FYI

**Comment by dominickng** [comment]
Thanks, we're aware of this issue and are looking to implement a fix.

Andre, do you know if TWAs have a specific referrer on Android?

**Comment by PEConn** [comment]
I believe the referer should be something like `android-app://com.mytwa.app` or something similar.

**Comment by andreban** [comment]
What @PEConn said, where`com.mytwa.app` is the package-name for the TWA.

**Comment by rockeynebhwani** [comment]
@PEConn / @andreban  - We rely on this to separate our mWeb vs TWA analytics. Since when this is broken? This was working on custom tabs client?

**Comment by andreban** [comment]
@rockeynebhwani this seems to be the case on Chrome OS only. On Android, the referrer is working as intended.

**Comment by alex2844** [comment]
@andreban yes, it is the case on Chrome OS only. Everything is ok with Android.

**Comment by rockeynebhwani** [comment]
@andreban - Did anything change in Chrome 81? Referrer post Chrome 81 is "android-app://com.app.george/" for our app before Chrome 80, it should be "android-app://com.app.george"

A '/' has appeared at the end.

**Comment by andreban** [comment]
@rockeynebhwani is this on Chrome OS only or on Android as well?

**Comment by rockeynebhwani** [comment]
@andreban - This is for Android. I just checked for our site, we have neither "android-app://com.app.george/" OR "android-app://com.app.george" as referrer for 'Chrome OS' visits

**Comment by rockeynebhwani** [comment]
@andreban - We are seeing theses referrers in Adobe Analytics but these are not appearing in Google Analytics at all. Are you aware of anything which will stop flowing these referrers in Google Analytics?

**Comment by andreban** [comment]
@rockeynebhwani Sorry, I got confused as this issue is about the Chrome OS referrer. I was not aware of the change. @PEConn may know more.

Having said that, I do think the change is for the better - It allows developers to check for the referrer using `referrer.startsWith('android-app://com.app.george/')` instead of `referrer === 'android-app://com.app.george'`, allowing extra information to be added to referrer without breaking the check (without the ending `/`, the `startsWith` check is broken, as an app called `'android-app://com.app.georges'` would be included.

I'd recommend creating a separate issue for this if you have further questions / discussion, as this is off-topic from the original issue.

**Comment by rockeynebhwani** [comment]
@andreban  - It's not a problem as we managed to change our segment in Adobe but we didn't see any announcement about this in Chrome 81 and it took some time for us to figure this out... Regarding Google Analytics issue, we are still investigating and if we need help, I will create a new issue.

**Comment by gstepniewski-google** [comment]
Closing due to no updates in 5+ years
